x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/search/SearchApplicationTemplate.java

```diff
@@ -29,59 +31,72 @@
 public class SearchApplicationTemplate implements ToXContentObject, Writeable {
     private final Script script;
 
+    private static final ParseField TEMPLATE_SCRIPT_FIELD = new ParseField("script");
+    public static final ParseField DICTIONARY_FIELD = new ParseField("dictionary");
+
+    private static final ConstructingObjectParser<SearchApplicationTemplate, Void> PARSER = new ConstructingObjectParser<>(
+        "search_template",
+        p -> new SearchApplicationTemplate((Script) p[0], (TemplateParamValidator) p[1])
+    );
+
+    static {
+        PARSER.declareObject(optionalConstructorArg(), (p, c) -> Script.parse(p, Script.DEFAULT_TEMPLATE_LANG), TEMPLATE_SCRIPT_FIELD);
+        PARSER.declareObject(optionalConstructorArg(), (p, c) -> {
```

Comment:
Some reordering done. We declare the `dictionary` tag and parsing in the top-level object (`SearchApplicationTemplate`), and let `TemplateParamValidator` do the (de)serialization inside the `dictionary` tag.